catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/common/TreUtility.java

```diff
@@ -55,4 +61,61 @@ public static Serializable getTreValue(TreGroup tre, String key) {
         }
         return Optional.empty();
     }
+
+    public static Integer parseInteger(Tre tre, String fieldName) {
+        Serializable value = TreUtility.getTreValue(tre, fieldName);
+        if (value instanceof String) {
+            return Integer.valueOf((String) value);
+        } else {
+            return null;
+        }
+    }
+
+    public static Float parseFloat(Tre tre, String fieldName) {
+        Serializable value = TreUtility.getTreValue(tre, fieldName);
+        if (value instanceof String) {
+            return Float.valueOf((String) value);
+        } else {
+            return null;
+        }
+    }
+
+    public static String parseString(Tre tre, String fieldName) {
+        Serializable value = TreUtility.getTreValue(tre, fieldName);
+        if (value instanceof String) {
+            return (String) value;
+        } else {
+            return null;
+        }
+    }
+
+    public static Boolean parseYNBoolean(Tre tre, String fieldName) {
+        Serializable value = TreUtility.getTreValue(tre, fieldName);
+        if (value instanceof String) {
+            String valueStr = (String) value;
+            return valueStr.equalsIgnoreCase("Y");
+        } else {
+            return null;
+        }
+    }
+
+    public static Date parseDate(Tre tre, String fieldName) {
+        Serializable value = TreUtility.getTreValue(tre, fieldName);
+        if (value instanceof String) {
+            String dateStr = (String) value;
+            SimpleDateFormat sdf = new SimpleDateFormat(TRE_DATE_FORMAT);
+            sdf.setTimeZone(TimeZone.getTimeZone("GMT"));
+            try {
+                return sdf.parse(dateStr);
+            } catch (ParseException e) {
+                LOGGER.debug("Unable to parse date {} according to format {}",
+                        dateStr,
+                        TRE_DATE_FORMAT,
+                        e);
+                return null;
+            }
+        } else {
+            return null;
```

Comment:
optional: remove the two 'return null' statements and the else-block, replace with a single 'return null' and the end of the method.